### DISCUSSION TITLE
如何以nodeport方式暴露端口

### DISCUSSION THREAD

**Discussion by codering** [emqx/emqx-operator#521] [Q&A] [answered]
我用的 operator v2版本，设置了 service的Type，并没有生效。
<img width="398" alt="image" src="https://user-images.githubusercontent.com/2271825/207010617-300918ec-6743-4c56-b762-c82b9449cf52.png">

**Comment by Rory-Z** [comment, marked as answer]
我使用 2.0.2 版本的 EMQX Operator，部署 NodePort 类型的 service 是正常的
```
$ kubectl get emqx emqx -o json | jq '.spec.dashboardServiceTemplate.spec.type,.spec.listenersServiceTemplate.spec.type'
"NodePort"
"NodePort"

$ kubectl get svc                                             
NAME             TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)           AGE
emqx-dashboard   NodePort    10.104.197.110   <none>        18083:32102/TCP   95s
emqx-headless    ClusterIP   None             <none>        4370/TCP          95s
emqx-listeners   NodePort    10.100.28.120    <none>        1883:30414/TCP    52s
kubernetes       ClusterIP   10.96.0.1        <none>        443/TCP           5d23h
```

可以分享一下 EMQX Operator Pod 的 log 么？ 看看有没有报错

**Reply by codering** [reply]
恩，看了日志，是端口冲突导致失败了，🙏

**Reply by codering** [reply]
顺便问下，我看默认replicant的数量是3，我可以设置为0吗？作为简单测试可以只用一个core吗？

**Reply by Rory-Z** [reply]
可以的

**Reply by codering** [reply]
感觉可以加一个Replicant的enabled配置更方便些

**Reply by Rory-Z** [reply]
`.spec.replicantTemplate.relicant = 0` 在某种意义上就是 disable，在 EMQX 5.0 中，我们更鼓励用户使用新的 Mria 集群模式（即配置了 relicant 节点与 code 节点），我们已经在一个由 3 个core 节点与 20 个 replicant 节点组成的集群上测试了一亿的客户端链接。
